### ISSUE TITLE
There is no in-depth documentation how to use AWS-SDK v3 in Lambda

### ISSUE THREAD

**Issue by Quutti** [aws/aws-sdk-js-v3#5354]
### Describe the issue

Currently we see issues under issue tracking that state i.e. missing dependency when using newest AWS-SDK client package version inside the lambda runtime (node18).

After reading many these issues i can see these two patterns appearing:

**Developers are told to exclude the @aws-sdk packages from the application itself and fully rely on the default (3.188) packages inside lambda runtime**
In my opinion this has massive problem as developers cannot rely on the latest documentation and the latest typings during the development phase. I.e. in our project we had situation where the LambdaClient Invoke Payload didn't introduce transformToString method on the default lambda client packages running on runtime. And ofc the latest lambda client package typings told us there is that kind of thing. 
https://github.com/aws/aws-sdk-js-v3/issues/4972#issuecomment-1642384809

**Should we then bundle the @aws-sdk packages inside the application itself?**
This has even bigger issues, as for example lib-dynamodb package died because it was screaming about missing dependency for \@smithy/smithy-client, and rds-client screaming for missing dependency tslib. Reading issue tracker it was stated that this happens because the inconsistency between the version within lambda runtime and the versions bundled with the application. Seems also pretty dangerous.

TLDR; it would be awesome to get a comprehensive guideline for this manner from the AWS-SDK team itself. Now it feels like developers are left to find the breadcrumbs from all over the place, and that makes things too anxious. And also, personally, it is starting to feel like i cannot rely on any approach fully.

Few related issues:
- https://github.com/aws/aws-sdk-js-v3/issues/4814#issuecomment-1587651761
- https://github.com/aws/aws-sdk-js-v3/issues/4075#issuecomment-1297933078 (fixed on newer package than the one running in lambda runtime)

### Links

Doesn't affect on any current documentation item. But requests new one.

**Comment by Quutti** [comment]
https://github.com/aws/aws-sdk-js-v3/issues/4345

And this one also as an addition - it states that the latest version should be used to mitigate this kind of bugs.

**Comment by RanVaknin** [comment]
Hi @Quutti ,

Thanks for opening this documentation issue. 

The SDK team aims to own only SDK specific documentation. However, we understand that a large subset of the SDK consumers are coming from the SDK vended through Lambda. The Lambda team has several official blogposts and docs on how to deploy or bundle your SDK for lambda:
https://aws.amazon.com/blogs/compute/optimizing-node-js-dependencies-in-aws-lambda/
https://aws.amazon.com/blogs/compute/node-js-18-x-runtime-now-available-in-aws-lambda/

Furthermore, the two issues you link are not specific to Lambda (or at least at a quick review they don't seem to be), so Im not sure how these two issues are tied to this documentation request. 

>I.e. in our project we had situation where the LambdaClient Invoke Payload didn't introduce transformToString method on the default lambda client packages running on runtime. And ofc the latest lambda client package typings told us there is that kind of thing.

If I understood you correctly, this could have been solved by bundling the latest version of the SDK and deploying it to Lambda, or using something like a Lambda layer. 

>Should we then bundle the @aws-sdk packages inside the application itself? 

You mean deploying your application with your `node_modules` file?  If so, it's possible, but I personally wouldn't recommend this as a solution. That would blow up the size of your lambda and might affect cold start time, and / or dependency issues like you have mentioned. 

With regards to issues that occurred with the migration to the `@smithy` namespace - the issues I have seen were largely regarding version mismatch, rather than a behavior unique to Lambda. So I don't think they are correlated.

As shown in one of the blogpost I shared, the best solution would be to bundle and minify your application with ESBuild.

If those resources were not what you are looking for, please write us the specific problem you are trying to solve so we can boil this down to a more actionable task.

Thanks again,
Ran~
